### ISSUE TITLE
8.14.0 environment variables stopped working

### ISSUE THREAD

**Issue by snevas** [elastic/logstash#16199]
**Logstash information**:

Please include the following information:

1. Logstash version (e.g. `bin/logstash --version`): 8.14.0
2. Logstash installation source (e.g. built from source, with a package manager: DEB/RPM, expanded from tar or zip archive, docker): from debian package
3. How is Logstash being run (e.g. as a service/service manager: systemd, upstart, etc. Via command line, docker/kubernetes): in docker with environment variable set

**Plugins installed**: (`bin/logstash-plugin list --verbose`): none

**JVM** (e.g. `java -version`): 17, but using packaged version

**OS version** (`uname -a` if on a Unix-like system): Debian 12

**Description of the problem including expected versus actual behavior**:
8.13.4 parsed the variable set in a pipeline to it's value, 8.14.0 doesn't

**Steps to reproduce**:
Upgrade/downgrade with same config (setting queue.max_bytes through environment variable)

Please include a *minimal* but *complete* recreation of the problem,
including (e.g.) pipeline definition(s), settings, locale, etc.  The easier
you make for us to reproduce it, the more likely that somebody will take the
time to look at it.

 1. Run 8.13.4, logstash starts
 2. Run 8.14.0, logstash breaks
 3. Revert back, logstash starts

**Provide logs (if relevant)**:
```
{"level":"ERROR","loggerName":"logstash.config.sourceloader","timeMillis":1717662013959,"thread":"Agent thread","logEvent":{"message":"Could not fetch all the sources","exception":"ArgumentError","message":"Unknown bytes value '${LOGSTASH_QUEUE_SIZE}'","backtrace":["/usr/share/logstash/logstash-core/lib/logstash/util/byte_value.rb:53:in `multiplier'","/usr/share/logstash/logstash-core/lib/logstash/util/byte_value.rb:33:in `parse'","/usr/share/logstash/logstash-core/lib/logstash/settings.rb:683:in `coerce'","/usr/share/logstash/logstash-core/lib/logstash/settings.rb:366:in `set'","/usr/share/logstash/logstash-core/lib/logstash/settings.rb:141:in `set_value'","/usr/share/logstash/logstash-core/lib/logstash/settings.rb:161:in `block in merge'","org/jruby/RubyHash.java:1610:in `each'","/usr/share/logstash/logstash-core/lib/logstash/settings.rb:161:in `merge'","/usr/share/logstash/logstash-core/lib/logstash/settings.rb:171:in `merge_pipeline_settings'","/usr/share/logstash/logstash-core/lib/logstash/config/source/multi_local.rb:36:in `block in pipeline_configs'","org/jruby/RubyArray.java:2803:in `map'","/usr/share/logstash/logstash-core/lib/logstash/config/source/multi_local.rb:34:in `pipeline_configs'","/usr/share/logstash/logstash-core/lib/logstash/config/source_loader.rb:76:in `block in fetch'","org/jruby/RubyArray.java:2798:in `collect'","/usr/share/logstash/logstash-core/lib/logstash/config/source_loader.rb:75:in `fetch'","/usr/share/logstash/logstash-core/lib/logstash/agent.rb:179:in `converge_state_and_update'","/usr/share/logstash/logstash-core/lib/logstash/agent.rb:117:in `execute'","/usr/share/logstash/logstash-core/lib/logstash/runner.rb:439:in `block in execute'","/usr/share/logstash/vendor/bundle/jruby/3.1.0/gems/stud-0.0.23/lib/stud/task.rb:24:in `block in initialize'"]}}
```

**Comment by andsel** [comment]
Hi @snevas thank't for reporting it.
I was looking to reproduce this error, so I tried:
```sh
docker run -e QUEUE_MAX_BYTES=1024mb -ti --rm docker.elastic.co/logstash/logstash:8.14.0 -e 'input { stdin {}} output { stdout { codec => "rubydebug" } }'
```

and started as expected.

Please could you provide an easy way to reproduce this issue? A docker command line would be ideal-

**Comment by hendry-lim** [comment]
Facing the same issue with Logstash `8.14.0` Docker image. I defined `LS_PIPELINE_DIR` env var, encountered this error in `8.14.0`. Working fine in `8.13.x`.

`LS_PIPELINE_DIR=/usr/share/logstash/conf.d`

`[2024-06-06T00:16:27,194][INFO ][logstash.config.source.local.configpathloader] No config files found in path {:path=>"/usr/share/logstash/${LS_PIPELINE_DIR}/pipeline_udp_output/*.logstash.conf"}`

```
# pipelines.yml
- pipeline.id: pipeline_udp_output
  path.config: "${LS_PIPELINE_DIR}/pipeline_udp_output/*.logstash.conf"
```

**Comment by snevas** [comment]
pipelines.yml
```
---
- pipeline.id: "test"
  config.string: >
    input { stdin {}} output { stdout { codec => "rubydebug" } }
  queue.max_bytes: "${LOGSTASH_QUEUE_SIZE}"
```

Broken:
`docker run --rm -it -e LOGSTASH_QUEUE_SIZE=128m -v ./pipelines.yml:/usr/share/logstash/config/pipelines.yml docker.elastic.co/logstash/logstash:8.14.0`

Okay:
`docker run --rm -it -e LOGSTASH_QUEUE_SIZE=128m -v ./pipelines.yml:/usr/share/logstash/config/pipelines.yml docker.elastic.co/logstash/logstash:8.13.4`

**Comment by jsvd** [comment]
@snevas until this is solved, is setting the queue size globally for logstash through the env var a valid workaround for you?

Either setting it directly:

```
❯ docker run -it -e "QUEUE_TYPE=persisted" -e "QUEUE_MAX_BYTES=99999999b" docker.elastic.co/logstash/logstash:8.14.0 --debug | grep max_bytes
2024/06/06 16:02:36 Setting 'queue.max_bytes' from environment.
[2024-06-06T16:02:42,231][DEBUG][logstash.runner          ] *queue.max_bytes: 99999999 (default: 1073741824)
[2024-06-06T16:02:42,231][DEBUG][logstash.runner          ] dead_letter_queue.max_bytes: 1073741824
```
Or loading it through logstash.yml instead of pipelines.yml:
```
❯ cat logstash.yml 
queue.type: persisted
queue.max_bytes: "${LOGSTASH_QUEUE_SIZE}"

❯ docker run -e LOGSTASH_QUEUE_SIZE=99999999b --rm -it -v ./logstash.yml:/usr/share/logstash/config/logstash.yml docker.elastic.co/logstash/logstash:8.14.0  --debug  | grep max_bytes
[2024-06-06T16:01:23,115][DEBUG][logstash.runner          ] *queue.max_bytes: 99999999 (default: 1073741824)
[2024-06-06T16:01:23,115][DEBUG][logstash.runner          ] dead_letter_queue.max_bytes: 1073741824
```

**Comment by snevas** [comment]
I will keep logstash on 8.13 until it's patched in 8.14 as a workaround. Thanks nonetheless @jsvd

**Comment by jsvd** [comment]
This has been fixed in 8.14.1 which has been released last week: https://www.elastic.co/guide/en/logstash/8.14/logstash-8-14-1.html
